server/src/com/google/refine/Refine.java

```diff
@@ -494,12 +494,42 @@ public void actionPerformed(ActionEvent e) {
     
     private void openBrowser() {
         if (!Desktop.isDesktopSupported()) {
-            logger.warn("Java Desktop class not supported on this platform.  Please open %s in your browser",uri.toString());
+            try {
+                openBrowser2();
+            } catch (IOException e) {
+                logger.error("Failed to open browser. Please open %s in your browser",uri.toString());
+            }
+        } else {
+            try {
+                Desktop.getDesktop().browse(uri);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
         }
-        try {
-            Desktop.getDesktop().browse(uri);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
+    }
+
+    private void openBrowser2() throws IOException {
+        logger.info("Opening browser to {}", uri.toString());
+        String os = System.getProperty("os.name").toLowerCase();
+        Runtime runtime = Runtime.getRuntime();
+        if(os.contains("win")) { // If we are on Windows
```

Comment:
Perhaps OS detection would be a bit cleaner if we used a library (which we already rely on): https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/SystemUtils.html#IS_OS_WINDOWS